### PR TITLE
remove things Haskell has that Elm doesn't

### PR DESCRIPTION
**Pull request by BrianHicks**
I don't have Gedit to test with (I'm on Mac, and consequently may have broken some references unawares) but I've removed some things that Elm doesn't have.

### CODE REVIEW

File: gedit/elm.lang [hunk 1]
```diff
@@ -41,23 +39,16 @@
...
       </include>
     </context>
 
-    <context id="pragma" style-ref="preprocessor">
-      <start>\{-#</start>
-      <end>#-\}</end>
-    </context>
-
     <context id="variable" style-ref="variable">
-      <match>\b[a-z_][0-9a-zA-Z_'#]*</match>
```
Comment (BrianHicks) [inline gedit/elm.lang:56]: `'` and `#` aren't valid Elm identifier characters.

**Comment by BrianHicks** [comment]
In addition to the escapes defined here, you can get a literal character in an escape sequence. For example `"\u{0041}"` evaluates to `"A"`. I wasn't sure how to add this! Note that the escape must be padded to at least 4 digits.

**Comment by vsoch** [comment]
Testing locally!

**Comment by vsoch** [comment]
Okay looks like there is one issue:

```
(gedit:30961): GtkSourceView-WARNING **: 11:39:40.509: Failed to load '/usr/share/gtksourceview-3.0/language-specs/elm.lang': invalid context reference “elm:octal”
```

**Comment by vsoch** [comment]
So we can either add back in:

```
    <context id="octal" style-ref="octal">
      <match>0[oO][0-7]+</match>
    </context>
```
or remove the reference to octal. I think the latter (removing the reference) would be correct, since you removed the context block?

**Comment by BrianHicks** [comment]
yep, that syntax doesn't exist in Elm. Sorry for the error! I don't have a way to test but I thought knowing more of the syntax might make this easy—apparently not! I'm happy to advise from here if you want to continue on it, but I think me doing the work is going to be too hit-or-miss. Sorry for the noise!

**Comment by vsoch** [comment]
Hey that's what I'm here for! I'll test every change you make, and show you a screenshot of the rendering for a sanity check.

**Comment by vsoch** [comment]
This LGTM

image

**Review by vsoch — APPROVED** [review]

**Comment by BrianHicks** [comment]
hooray! Just looking at that, it might be helpful to make a keyword for "type alias", if possible, and perhaps `->`, which is only used in type signatures and case expressions.

**Comment by vsoch** [comment]
@BrianHicks would you like to take a shot? There is definitely no rush!

**Comment by BrianHicks** [comment]
I probably won't, to be honest. I have no idea about the syntax for these: type alias has a space, and I don't know if -> needs any kind of escaping. I appreciate your offer of screenshots, though!

**Pull request merged**